src/Data/Vec/Functional.agda

```diff
@@ -0,0 +1,75 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Vectors defined via index notation
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Vec.Functional where
+
+open import Data.Fin using (Fin; zero; suc)
+open import Data.Nat using (ℕ; zero; suc; _+_)
+open import Data.Product using (Σ; ∃; _×_; _,_; proj₁; proj₂)
+open import Data.Vec as V using (Vec)
+open import Function
+open import Level using (Level)
+
+infixr 5 _∷_
+
+private
+  variable
+    a b : Level
+    A : Set a
+    B : Set b
+    m n : ℕ
+
+Vector : Set a → ℕ → Set a
+Vector A n = Fin n → A
+
+map : (A → B) → (Vector A n → Vector B n)
+map f v = f ∘ v
```

Comment:
I think best if we stuck to the convention of using `xs`, `ys` etc. for collections of objects. `v` is usually reserved for a single value (e.g. in membership proofs).